app/assets/javascripts/poller.js

```diff
@@ -5,15 +5,37 @@
     init: function() {
       this.$poller = $('.js-poller');
       this.$messageForm = $('.js-message-form');
+      this.pollingTimeout = 1000 * 60 * 60; // 1 hour
 
       if (this.$poller.length && this.$messageForm.length) {
         this.setLast(this.timestamp());
+        this.bindEvents();
         this.start();
       }
     },
 
+    pageVisibilitySupported: function() {
+      return 'hidden' in document ? true : false;
+    },
+
+    bindEvents: function() {
+      if (this.pageVisibilitySupported()) {
+        $(document).on('visibilitychange', this.handleVisibilityChange.bind(this));
+      }
+    },
+
+    handleVisibilityChange: function() {
+      return document['hidden'] ? this.stop() : this.start();
```

Comment:
This feels like a duplicate of line 18 – should we be using the same method to check for support?